kpi/utils/private_storage.py

```diff
@@ -40,6 +44,13 @@ def superuser_or_username_matches_prefix(private_file):
     if private_file.relative_name.startswith(
         '{}/'.format(user.username)
     ):
+        if user.pk == -1:
+            filename_regex = r'AnonymousUser/exports/(a[^/]*)/.*'
+            match = re.search(filename_regex, private_file.relative_name)
+            if match:
+                uid = match.groups()[0]
+                a = Asset.objects.get(uid=uid)
```

Comment:
I don't think we need to all fields for that. Let's load only needed fields (especially avoid `content`) .